dbms/src/Flash/Planner/PhysicalPlan.cpp

```diff
@@ -54,8 +54,18 @@ void fillOrderForListBasedExecutors(DAGContext & dag_context, const PhysicalPlan
     auto & list_based_executors_order = dag_context.list_based_executors_order;
     PhysicalPlanVisitor::visitPostOrder(root_node, [&](const PhysicalPlanNodePtr & plan) {
         assert(plan);
-        if (plan->isRecordProfileStreams())
-            list_based_executors_order.push_back(plan->execId());
+        if (plan->isTiDBOperator())
+        {
+            if (plan->tp() == PlanType::TableScan)
+            {
+                auto physical_table_scan = std::static_pointer_cast<PhysicalTableScan>(plan);
```

Comment:
why not dynamic_pointer_cast?

Reply:
Because plan must be `PhysicalTableScan`.
https://github.com/pingcap/tiflash/pull/5186
https://en.cppreference.com/w/cpp/memory/shared_ptr/pointer_cast

Reply:
> Because plan must be `PhysicalTableScan`. #5186 https://en.cppreference.com/w/cpp/memory/shared_ptr/pointer_cast

Yes, but IMO the downcast is not safe, if we use dynamic cast, we can ensure the safety of cast.

Reply:
Yes, if plan->tp() == PlanType::TableScan, plan must be PhysicalTableScan.

Reply:
> 

if not, this must be a stupid bug.